tls/s2n_psk.h

```diff
@@ -33,6 +33,18 @@ typedef enum {
     S2N_PSK_DHE_KE,
 } s2n_psk_key_exchange_mode;
 
+struct s2n_psk_identity{
+    uint8_t *identity;
+    size_t identity_length;
+};
+
+struct s2n_pre_shared_key {
+    struct s2n_psk_identity identity;
+    uint8_t *secret;
+    size_t secret_len;
+    s2n_hmac_algorithm hmac;
+};
```

Comment:
Why are we duplicating struct s2n_psk?

Reply:
We want to give customers an external struct to set up their psks. We can't use the previously defined s2n_psk because it contains information that the customer does not need to set, like the early_secret variable. So my pr creates a new struct that only contains what the customer needs to provide the api. Then, the call to s2n_set_connection_psks() takes the external psks and turns them into our internal psk representation.